File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, arr: List[int]) -> int:
        d={} 
        for i in arr:
            if i not in d: 
                d[i]=1 
            else: 
                d[i]+=1 
        for k,v in d.items():
            if v==1:
                return k

File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer,Integer> hm=new HashMap<>();
        for(int ele:nums){
            if(hm.containsKey(ele)){
                hm.put(ele,hm.get(ele)+1);
            }else{
                hm.put(ele,1);
            }
        }
        for (Map.Entry<Integer, Integer> entry : hm.entrySet()) {
            if (entry.getValue() == 1) {
                return entry.getKey();
            }
        }
        return -1;
    }
}